mod/state-transition/pkg/core/helpers_test.go

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2024, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package core_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	corestore "cosmossdk.io/core/store"
+	"cosmossdk.io/log"
+	"cosmossdk.io/store"
+	"cosmossdk.io/store/metrics"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/berachain/beacon-kit/mod/consensus-types/pkg/types"
+	engineprimitives "github.com/berachain/beacon-kit/mod/engine-primitives/pkg/engine-primitives"
+	"github.com/berachain/beacon-kit/mod/node-core/pkg/components"
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/common"
+	"github.com/berachain/beacon-kit/mod/state-transition/pkg/core/mocks"
+	"github.com/berachain/beacon-kit/mod/storage/pkg/beacondb"
+	"github.com/berachain/beacon-kit/mod/storage/pkg/db"
+	"github.com/berachain/beacon-kit/mod/storage/pkg/encoding"
+	dbm "github.com/cosmos/cosmos-db"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestExecutionEngine_VerifyAndNotifyNewPayload(t *testing.T) {
```

Comment:
@nidhi-singh02 I am not sure I would keep this test.
I understand that as we create the mock, we want to be sure it works as expected, but I don't think we need to check over and over with the CI that mock is fine. After all the mock itself is not production code.
I would recommend dropping this (or if we really want to keep it because, say, we already do it for other mocks in the codebase, consider moving it to the `mock` folder)?

Reply:
There was a test for VerifyAndNotifyNewPayload, I thought we wanted to test the same via mock. I understand the we are calling `VerifyAndNotifyNewPayload` on mock engine, but not with a mock request, instead with the actual 
NewPayloadRequest.

`mock` folder does not contain code other then those generated by mockery, so doesn't seem the right place to keep the tests. I could drop it if we don't need it.